types/src/connectors/api.ts

```diff
@@ -0,0 +1,40 @@
+export type ConnectorsAPIErrorType =
+  | "authorization_error"
+  | "not_found"
+  | "internal_server_error"
+  | "unexpected_error_format"
+  | "unexpected_response_format"
+  | "unknown_connector_provider"
+  | "invalid_request_error"
+  | "connector_not_found"
+  | "connector_configuration_not_found"
+  | "connector_update_error"
+  | "connector_update_unauthorized"
+  | "connector_oauth_target_mismatch"
+  | "slack_channel_not_found"
+  | "connector_rate_limit_error";
+
+export type ConnectorsAPIError = {
+  type: ConnectorsAPIErrorType;
+  message: string;
+};
+
+export type ConnectorsAPIErrorResponse = {
+  error: ConnectorsAPIError;
+};
+
+export type ConnnectorsAPIErrorWithStatusCode = {
+  api_error: ConnectorsAPIError;
+  status_code: number;
+};
+
+export type WithConnectorsAPIErrorReponse<T> = T | ConnectorsAPIErrorResponse;
+
+export function isConnectorsAPIError(obj: unknown): obj is ConnectorsAPIError {
+  return (
+    typeof obj === "object" &&
+    obj !== null &&
+    typeof (obj as ConnectorsAPIError).message === "string" &&
+    typeof (obj as ConnectorsAPIError).type === "string"
```

Comment:
Small shortcut here 😅.
It should have been:
```suggestion
    "message" in obj && obj.message === "string" &&
    "type" in obj && obj.type === "string"
```